src/components/subcomponents/upload.js

```diff
@@ -0,0 +1,212 @@
+          setFiles([...files]);
+        });
+      }
+    });
+  };
+
+  const classes = useStyles();
+
+  return (
+    <Grid
+      container
+      justify="space-between"
+      alignItems="center"
+      alignContent="center"
+      spacing={1}
+      className={classes.root}
+      style={{ width: "inherit", margin: "2rem" }}
+    >
+      <Grid item xs={12}>
+        <Typography variant="h6">Attach Files</Typography>
+      </Grid>
+      <Grid item xs={12}>
+        <Typography variant="body1">[Upload instructions Here]</Typography>
+      </Grid>
+      <Grid item xs={12}>
+        <input
+          className={classes.input}
+          type="file"
+          id="upload-field"
+          name="uploadFiles"
+          multiple
+          onChange={onAddFile}
+        />
+        <label htmlFor="upload-field">
+          <Button variant="outlined" color="primary" component="span">
+            Add Files
+          </Button>
+        </label>
+      </Grid>
+      <Grid item xs={12}>
+        <Grid container spacing={2}>
+          {files.map((file, index) => {
+            return (
+              <Grid item key={index}>
+                <Paper className={classes.attachment}>
+                  <IconButton
+                    className={classes.attachDel}
+                    color="secondary"
+                    component="span"
+                    size="small"
+                    onClick={e => {
+                      e.preventDefault()
+                      onDelete(file)
+                    }}
+                  >
+                    <HighlightOffIcon />
+                  </IconButton>
+                  <AttachFileIcon
+                    className={classes.attachIco}
+                    fontSize="large"
+                    disabled
+                  />
+                  <Typography variant="body1" className={classes.overflow}>
+                    {file.data.name}
+                  </Typography>
+                  <Typography variant="body1">
+                    {(file.data.size / 1000000).toFixed(2) + "MB"}
+                  </Typography>
+                  {(() => {
+                    switch (file.upload) {
+                      case "UPLOADING":
+                        return <LinearProgress className={classes.loading} />;
+                      case "UPLOADED":
+                        return (
+                          <LinearProgress
+                            variant="determinate"
+                            value={100}
+                            className={classes.loading}
+                          />
+                        );
+                      case "FAILED":
+                        return (
+                          <Typography variant="body2" color="secondary">
+                            Upload Failed
+                          </Typography>
+                        );
+                      default:
+                        return null;
+                    }
+                  })()}
```

Comment:
this self invoking function is a bit hidden, can we create a `<FileUploadProgress>` component that handles this outside of this one huge component

Reply:
Yep I can move it out